src/main/java/jenkins/branch/OrganizationFolder.java

```diff
@@ -0,0 +1,241 @@
+                            }
+                            sources = null; // make sure complete gets called just once
+                            return branchSources;
+                        }
+                        @Override
+                        public void addAttribute(String key, Object value) throws IllegalArgumentException, ClassCastException {
+                            throw new IllegalArgumentException();
+                        }
+                        @Override
+                        public void complete() throws IllegalStateException, InterruptedException {
+                            MultiBranchProject<?,?> existing = observer.shouldUpdate(projectName);
+                            if (existing != null) {
+                                PersistedList<BranchSource> sourcesList = existing.getSourcesList();
+                                sourcesList.clear();
+                                sourcesList.addAll(createBranchSources());
+                                return;
+                            }
+                            if (!observer.mayCreate(projectName)) {
+                                listener.getLogger().println("Ignoring duplicate child " + projectName);
+                                return;
+                            }
+                            for (MultiBranchProjectFactory factory : projectFactories) {
+                                MultiBranchProject<?, ?> project;
+                                try {
+                                    project = factory.createProject(OrganizationFolder.this, projectName, sources, Collections.<String,Object>emptyMap(), listener);
+                                } catch (InterruptedException x) {
+                                    throw x;
+                                } catch (Exception x) {
+                                    x.printStackTrace(listener.error("Failed to create a subproject " + projectName));
+                                    continue;
+                                }
+                                if (project != null) {
+                                    project.getSourcesList().addAll(createBranchSources());
+                                    observer.created(project);
+                                    project.scheduleBuild();
+                                    break;
+                                }
+                            }
+                        }
+                    };
+                }
+                @Override
+                public void addAttribute(String key, Object value) throws IllegalArgumentException, ClassCastException {
+                    throw new IllegalArgumentException();
+                }
+            });
+        }
+    }
+
+    @Override
+    public List<SCMSource> getSCMSources() {
+        return Collections.emptyList(); // irrelevant unless onSCMSourceUpdated implemented
```

Comment:
Not used in this plugin, but I see it useful outside (a way to get the list of repositories managed by this `OrganizationFolder`). I think returning the list `SCMSource`s owned by the multiple `SCMNavigator`s is harmless (and possibly useful outside).